miner/worker.go

```diff
@@ -561,7 +561,8 @@ func (env *Work) commitTransactions(mux *event.TypeMux, txs *types.TransactionsB
 	gp := new(core.GasPool).AddGas(env.header.GasLimit)
 
 	var coalescedLogs vm.Logs
-	for {
+
+	for i := 0; ; i++ {
```

Comment:
This `i++` won't be good here. There are a few scenarios below when a transaction is discarded (gas price too low, doesn't fit into block, etc). These scenarios will all "continue looping" without processing the transaction, but the `i++` will increment nonetheless, producing gaps.

Shouldn't we use `env.tcount++` (incremented in L604) to keep track of the index of the currently executing transaction and completely drop `i`?